### DISCUSSION TITLE
有谁布置gRPC+Cloudflare成功了，给个配置文件模板参考下

### DISCUSSION THREAD

**Discussion by Geekvcn** [v2fly/v2ray-core#792] [Q&A]
我布置完后连CF不通，裸连龟速，WSS的话就一切正常

**Comment by Geekvcn** [comment]
成功了，SSL/TLS 加密模式为 完全 才可以，速度较websocket没提升

**Comment by CrazyBoyFeng** [comment]
速度提升应该不明显，流量大时可能反而感觉会慢。另外它更耗电。

grpc 是用 http2 承载的，理论上其会继承 h2 的大部分特性。

h2 可以简单看作是 ws 加多路复用加头部压缩。那么多路复用的优缺点也会体现在 h2 和 grpc 上：  
* 流量比较大时，h2 会消耗比较多的 cpu 在解压缩和拆帧上，从而相比于 ws 会显得慢。
* 流量比较小且并发连接比较多的情况下，h2 相比 ws 会节省出握手的时间，传输量也较少，因此会稍微显得快一点。
* 由于前述 h2 多路复用等特性，其比 ws 更耗电。流量越大越耗电。
* ws 只能以 http1.1 的 alpn 发起，存在被识别的可能性 #557。正常的现代网络程序均已兼容 h2。相比之下，在服务器支持 h2 的情况下使用 h2 会更加自然。

**Comment by CrazyBoyFeng** [comment]
我体验cf+grpc几个月。结论：
在不被干扰的情况下，稳定性不如cf+ws。原因是cf内部是http1.1，grpc流量被转换成http1.1传输。cf还搞了很多分割合并之类的“优化”，这就造成很多问题，断流的情况比较严重。如果有谁知道怎么解决，希望能分享一下。
不用cf只用grpc稳定性就很好。

目前cf的两个方案，ws和grpc都不是非常理想。ws有一定概率被识别，也容易被干扰，grpc断流。
如果只看网页，不玩游戏的话，grpc断流的影响不大，而且受干扰的情况较少。这种情况下可以用grpc。
